gitpod-io/gitpod
# Allow enabling/disabling Incremental Prebuilds in Project Settings

## Description


Allow enabling/disabling Incremental Prebuilds in Project Settings:

- [x] Add a new Project Settings menu containing:
  - [x] The existing Configuration page
  - [x] Another page with a "Enable Incremental Prebuilds" checkbox
- [x] If it's enabled for a Project, make new Prebuilds incremental
- [x] In the Prebuild page, show when a Prebuild is incremental
- [x] Drive-by: Improve formatting of Prebuild duration
- [x] Drive-by: Correctly show 'timed out' status in Prebuild Logs

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/6727

## How to test


1. Add a Project
2. Start a Prebuild, and wait for it to be successful
3. Only after it finished successfully, push a new commit to the same branch
4. This should trigger a new Prebuild, but the new Prebuild should not be incremental
5. Next, in the Project's Settings, enable "Use Incremental Prebuilds"
6. Push another commit to the same branch
7. The newly triggered Prebuild should be incremental

## Release Notes

```release-note
Allow enabling/disabling Incremental Prebuilds in Project Settings
Also improve Prebuild Logs UX (show incremental status, format duration nicely, correctly show 'timed out' status)
```

## Documentation


/uncc
